tests/capture/input_generators/test_attack_point_iterator.py

```diff
@@ -145,3 +147,247 @@ def test_attack_point_iterator_unrestricted_generator_all_args_len():
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(build_attack_points_iterator(config))
+    assert output == []
+
+
+def test_attack_point_iterator_repeat_zero_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 0,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == 0
+
+
+def test_attack_point_iterator_repeat_one():
+    config = {
+        "operation": "repeat",
+        "repetitions": 1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(iter(build_attack_points_iterator(config)))
+    assert output == list(
+        iter(build_attack_points_iterator(config["configuration"])))
+
+
+def test_attack_point_iterator_repeat_one_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == len(config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_two():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(iter(build_attack_points_iterator(config)))
+    assert output == list(
+        iter(build_attack_points_iterator(
+            config["configuration"]))) * config["repetitions"]
+
+
+def test_attack_point_iterator_repeat_two_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_three_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 3,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_infinite():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1]
```

Comment:
Please either have a longer list here or create a similar test with more values.

Reply:
There is the test with repetitions: -2 that has values: [1, 2]

Reply:
Thank you for that other test. But this test as it is written fails when at line 275 you add more values to the list. And it is a bug in the test itself because on line 288 you multiply by the number of iterations `stop_count`. I am ok here with line 286 having a hardcoded expected value. But the current unit-test is rather confusing (as in when I change something it breaks and I would be blaiming the implementation not the test).